cmd/swctl/cmd_status.go

```diff
@@ -34,23 +64,240 @@ func NewStatusCmd(cli Cli) *cobra.Command {
+			return err
+		}
+	}
 	return nil
 }
+
+func countConfig(baseVals []*kvscheduler.BaseValueStatus) configCounts {
+	var allVals []*kvscheduler.ValueStatus
+	for _, baseVal := range baseVals {
+		allVals = append(allVals, baseVal.Value)
+		allVals = append(allVals, baseVal.DerivedValues...)
+	}
+
+	var res configCounts
+	for _, val := range allVals {
+		switch val.State {
+		case kvscheduler.ValueState_INVALID, kvscheduler.ValueState_FAILED:
+			res.Err++
+		case kvscheduler.ValueState_MISSING:
+			res.Missing++
+		case kvscheduler.ValueState_PENDING:
+			res.Pending++
+		case kvscheduler.ValueState_RETRYING:
+			res.Retrying++
+		case kvscheduler.ValueState_UNIMPLEMENTED:
+			res.Unimplemented++
+		case kvscheduler.ValueState_CONFIGURED, kvscheduler.ValueState_DISCOVERED, kvscheduler.ValueState_OBTAINED, kvscheduler.ValueState_REMOVED, kvscheduler.ValueState_NONEXISTENT:
+			res.Ok++
+		}
+	}
+	return res
+}
+
+func cmpStatus(a, b statusInfo) bool {
+	greater := a.GetMode() > b.GetMode()
+	if !greater && a.GetMode() == b.GetMode() {
+		greater = a.GetName() > b.GetName()
+	}
+	return greater
+}
+
+func printStatusTable(out io.Writer, infos []statusInfo) {
+	table := tablewriter.NewWriter(out)
+	header := []string{
+		"Name", "Mode", "IP Address", "GPRC Port", "HTTP Port", "Status", "Configuration",
+	}
+	aleft := tablewriter.ALIGN_LEFT
+	acenter := tablewriter.ALIGN_CENTER
+	table.SetHeader(header)
+	table.SetAutoWrapText(false)
+	table.SetAutoFormatHeaders(true)
+	table.SetHeaderAlignment(tablewriter.ALIGN_LEFT)
+	table.SetColumnAlignment([]int{aleft, aleft, aleft, acenter, acenter, acenter, aleft})
+	table.SetCenterSeparator("")
+	table.SetColumnSeparator("")
+	table.SetRowSeparator("")
+	table.SetHeaderLine(false)
+	table.SetBorder(false)
+	table.SetTablePadding("\t")
+	for _, info := range infos {
+		row := []string{info.GetName(), compoModeString(info.GetMode())}
+		var clrs []tablewriter.Colors
+		if info.GetMode() == client.ComponentForeign {
+			clrs = []tablewriter.Colors{{}, {}}
+			for i := range header[2:] {
+				clrs = append(clrs, []int{tablewriter.FgHiBlackColor})
+				row = append(row, strings.Repeat("-", len(header[i+2])))
+			}
+			table.Rich(row, clrs)
+			continue
+		}
+		config := info.ConfigCounts.string()
+		configColor := info.ConfigCounts.color()
+		compoInfo := info.GetInfo()
+		grpcState := compoInfo.GRPCConnState.String()
+		var statusClr int
+		// gRPC state does not make sense for StoneWork itself
+		if info.GetMode() == client.ComponentStonework {
+			grpcState = strings.Repeat("-", len("Status"))
+			statusClr = tablewriter.FgHiBlackColor
+		}
+		row = append(row,
+			compoInfo.IPAddr,
+			strconv.Itoa(compoInfo.GRPCPort),
+			strconv.Itoa(compoInfo.HTTPPort),
+			grpcState,
+			config)
+		clrs = []tablewriter.Colors{
+			{}, {}, {}, {}, {}, {statusClr}, {configColor},
+		}
+		table.Rich(row, clrs)
+	}
+	table.Render()
+}
+
+func compoModeString(c client.ComponentMode) string {
```

Comment:
This should be implemented where the type is defined using `String() string` method.